### PR TITLE
docs(middleware.md): translated & proofread 'middleware.md'

### PR DESCRIPTION
**Pull request by lslxdx**
##### Checklist
- [x] documentation is changed or added

##### Description of change
'middleware.md' is translated by @lslxdx and proofread by @闷油瓶小张

mv to https://github.com/eggjs/egg/pull/784

### CODE REVIEW

File: docs/source/en/basics/middleware.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+We begin by writing a simple gzip middleware, to see how to write middleware
+
+```js
+const isJSON = require('koa-is-json');
+const zlib = require('zlib');
+
+function* gzip(next) {
+  yield next;
+
+  // convert the reaponse body to gzip after the completion of the execution of subsequent middleware
```
Comment (atian25) [inline docs/source/en/basics/middleware.md]: typo: `reaponse` -> `response`

File: docs/source/en/basics/middleware.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+  let body = this.body;
+  if (!body) return;
+  if (isJSON(body)) body = JSON.stringify(body);
+
+  // set gzip body, correct the reponse header
```
Comment (atian25) [inline docs/source/en/basics/middleware.md]: `reponse` -> `response`

File: docs/source/en/basics/middleware.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+}
+```
+
+You might find that the middleware's writing style in the framework is exactly the same as in Koa, so any middleware in Koa can be used directly by the framework.
+
+### Configuration
+
+Usually the middleware has its own configuration. In the framework, a complete middleware is including the configuration process. We agree that a middleware is a separate file placed in `app/middleware` directory, which needs a exports function that take two paramters:
+
+- options: the configuration field of the middleware, `app.config[${middlewareName}]` will be passed in by the frame
```
Comment (atian25) [inline docs/source/en/basics/middleware.md]: `by the framework` ?

File: docs/source/en/basics/middleware.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+```js
+const isJSON = require('koa-is-json');
+const zlib = require('zlib');
+
+module.exports = options => {
+  return function* gzip(next) {
+    yield next;
+
+    // convert the reaponse body to gzip after the completion of the execution of subsequent middleware
```
Comment (atian25) [inline docs/source/en/basics/middleware.md]: `reaponse`

File: docs/source/en/basics/middleware.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+    let body = this.body;
+    if (!body) return;
+
+    // support options.threshold
+    if (options.threshold && this.length < options.threshold) return;
+
+    if (isJSON(body)) body = JSON.stringify(body);
+
+    // set gzip body, correct the reponse header
```
Comment (atian25) [inline docs/source/en/basics/middleware.md]: `reponse`

File: docs/source/en/basics/middleware.md [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+    threshold: 1024, // skip response body which size is less than 1K
+  },
+};
+```
+
+** About Configuration fields and runtime environment configurations, see [Config](./config.md) chapter. **
+
+## Default Framework Middleware
+
+In addition to the application layer middleware is imported, the framework itself and other plug-ins will also import many middleware. All the config fields of these built-in middlewares can be modified by modifying the ones with the same name in the config file, for example [Framework Built-in Plugin](https://github.com/eggjs/egg/tree/master/app/middleware) uses a bodyParser middleware(the framework loader will change the file name separated by delimiters into the camel style), and we can add configs below in `config/confg.default.js` to modify the bodyParser:
```
Comment (atian25) [inline docs/source/en/basics/middleware.md]: typo: `config/confg.default.js` -> `config/config.default.js`

File: docs/source/en/basics/middleware.md (hunk 3 above)
Comment (lslxdx) [inline docs/source/en/basics/middleware.md]: 👍 You're right, it's my fault. 

Next time I will paste the doc to LibreOffice (any recommendation?) to find typos before commit it.

File: docs/source/en/basics/middleware.md (hunk 1 above)
Comment (lslxdx) [inline docs/source/en/basics/middleware.md]: typo 👍

File: docs/source/en/basics/middleware.md (hunk 2 above)
Comment (lslxdx) [inline docs/source/en/basics/middleware.md]: typo 👍

File: docs/source/en/basics/middleware.md (hunk 6 above)
Comment (lslxdx) [inline docs/source/en/basics/middleware.md]: typo 👍

**Comment by lslxdx** [comment]
Fixed and the revision is 1bdef05.

**Comment by lslxdx** [comment]
What should I do next to make this PR be merged?

**Comment by atian25** [comment]
@lslxdx just wait for other guys review

**Comment by fengmk2** [comment]
Please use github email as commit author email

**Comment by lslxdx** [comment]
@fengmk2 Oops, I forgot it!

Should I re-commit the doc and create a new PR?

**Comment by lslxdx** [comment]
@fengmk2 I've changed my committing name and email and then create a new PR (#784), would you please help to review it?

Thank you so much!

**Pull request closed without merging**